src/Kiota.Builder/Refiners/HttpRefiner.cs

```diff
@@ -0,0 +1,152 @@
+        CrawlTree(current, CapitalizeNamespacesFirstLetters);
+    }
+
+    private static void SetBaseUrlForRequestBuilderMethods(CodeElement current, string? baseUrl)
+    {
+        if (baseUrl is not null && current is CodeClass codeClass && codeClass.IsOfKind(CodeClassKind.RequestBuilder))
+        {
+            // Add a new property named BaseUrl and set its value to the baseUrl string
+            var baseUrlProperty = new CodeProperty
+            {
+                Name = BaseUrl,
+                Kind = CodePropertyKind.Custom,
+                Access = AccessModifier.Private,
+                DefaultValue = baseUrl,
+                Type = new CodeType { Name = BaseUrlName, IsExternal = true }
+            };
+            codeClass.AddProperty(baseUrlProperty);
+        }
+        CrawlTree(current, (element) => SetBaseUrlForRequestBuilderMethods(element, baseUrl));
+    }
+
+    private void RemoveUnusedCodeElements(CodeElement element)
+    {
+        if (!IsRequestBuilderClass(element) || IsBaseRequestBuilder(element) || IsRequestBuilderClassWithoutAnyHttpOperations(element))
+        {
+            var parentNameSpace = element.GetImmediateParentOfType<CodeNamespace>();
+            parentNameSpace?.RemoveChildElement(element);
+        }
+        CrawlTree(element, RemoveUnusedCodeElements);
+    }
+
+    private void AddPathParameters(CodeElement element)
+    {
+        var parent = element.GetImmediateParentOfType<CodeNamespace>().Parent;
+        while (parent is not null)
+        {
+            var codeIndexer = parent.GetChildElements(false)
+                .OfType<CodeClass>()
+                .FirstOrDefault()?
+                .GetChildElements(false)
+                .OfType<CodeMethod>()
+                .FirstOrDefault(static x => x.IsOfKind(CodeMethodKind.IndexerBackwardCompatibility));
+
+            if (codeIndexer is not null)
+            {
+                // Retrieve all the parameters of kind CodeParameterKind.Custom
+                var customParameters = codeIndexer.Parameters
+                    .Where(static param => param.IsOfKind(CodeParameterKind.Custom))
+                    .ToList();
+
+                // For each parameter:
+                foreach (var param in customParameters)
+                {
+                    // Create a new property of kind CodePropertyKind.PathParameters using the parameter and add it to the codeClass
+                    var pathParameterProperty = new CodeProperty
+                    {
+                        Name = param.Name,
+                        Kind = CodePropertyKind.PathParameters,
+                        Type = param.Type,
+                        Access = AccessModifier.Public,
+                        DefaultValue = param.DefaultValue,
+                        SerializationName = param.SerializationName,
+                        Documentation = param.Documentation
+                    };
+
+                    if (element is CodeClass codeClass)
```

Comment:
it seems to me that condition should be at the very beginning of the method, no point doing all that work if we cannot add anything in the end. (make sure you don't bypass the crawl tree though)